spec/models/transaction_spec.rb

```diff
@@ -8,4 +8,10 @@
   describe "enum" do
     it { should define_enum_for(:result).with_values(["success", "failed"]) }
   end
+
+  describe "validations" do
+    it { should validate_presence_of :credit_card_number }
+    it { should validate_presence_of :credit_card_expiration_date }
+    it { should validate_presence_of :result }
```

Comment:
Same as the above tests for status, where I think we need to test that it comes in only as either "success" or "failed"